ext/json/json_encoder.c

```diff
@@ -60,10 +60,15 @@ static inline void php_json_pretty_print_char(smart_str *buf, int options, char
 static inline void php_json_pretty_print_indent(smart_str *buf, int options, php_json_encoder *encoder) /* {{{ */
 {
 	int i;
+	int indent_length = 0;
 
-	if (options & PHP_JSON_PRETTY_PRINT) {
+	if (encoder->indent_str) {
+	    indent_length = ZSTR_LEN(encoder->indent_str);
```

Comment:
Why does this care about indent_length? Can't we just append indent_str unconditionally? I don't think we need to optimize for the indent=0 case.

Reply:
Because indent_length = 0 is basically a noop for the indent function. I have updated the function to be more clear about that scenario. 

Let me know if you still think it'd be better to not care about the `indent_length`!